Add geometric fields documentation (#71)
exasim-project/NeoFOAM: doc/finiteVolume/cellCentred/fields.rst

.. _fvcc_fields:

Fields (FVCC)

> =============

Reviewer: Could we get a class diagram at the top here, just to orient the reader?

Author: Yes, we should check if this works
https://github.com/mgaitan/sphinxcontrib-mermaid

.. warning::
    The API of the classes probably will change in the future as currently parallelization is not supported.

Cell centered fields
^^^^^^^^^^^^^^^^^^^^

The ``fvccVolField`` stores the field values at cell centers and along boundaries, providing essential data for constructing the DSL (Domain Specific Language). This functionality also includes access to mesh data, integrating closely with the computational framework.

> ``DomainField`` acts as the fundamental data container within this structure, offering both read and to the ``internalField`` and ``boundaryField`` provided by the ``DomainField``. The ``correctBoundaryConditions`` member function updates field's boundary conditions, which are specified at construction. The boundaryConditions do not hold the data but modify the ``DomainField`` or ``BoundaryField`` container.

Reviewer: ```suggestion
``DomainField`` acts as the fundamental data container within this structure, offering both read and write to the ``internalField`` and ``boundaryFields`` provided by the ``DomainField``. The ``correctBoundaryConditions`` member function updates the field's boundary conditions, which are specified at construction. The boundaryConditions do not hold the data but rather modify the ``DomainField`` or ``BoundaryField`` container.
```

Functionally, fvccVolField parallels several OpenFOAM classes such as ``volScalarField``, ``volVectorField``, and ``volTensorField``.

.. doxygenclass:: NeoFOAM::fvccVolField
    :members:
        field_,
        fvccVolField,
        internalField,
        boundaryField,
        correctBoundaryConditions

> Face centered fields

Reviewer: Here again is it Face Centred Fields or face centred fields?

^^^^^^^^^^^^^^^^^^^^

The ``fvccSurfaceField`` stores the field values at face centers and along boundaries, providing essential data for constructing the DSL (Domain Specific Language). This functionality also includes access to mesh data, integrating closely with the computational framework.

> ``DomainField`` acts as the fundamental data container within this structure, offering both read and to the ``internalField`` and ``boundaryField`` provided by the ``DomainField``. The ``correctBoundaryConditions`` member function updates field's boundary conditions, which are specified at construction. The boundaryConditions do not hold the data but modify the ``DomainField`` or ``BoundaryField`` container.

Reviewer: similar to above

Reviewer 2: could you elaborate?

Functionally, fvccVolField parallels several OpenFOAM classes such as ``surfaceScalarField``, ``surfaceVectorField``, and ``surfaceTensorField``.

> However, the internalField also contains the boundary values, so no branches (if) are required when loop over all cell faces. So the size of the internalField in the NeoFOAM and OpenFOAM is different.

Reviewer: ```suggestion
However, the internalField also contains the boundary values, so no branches (if) are required when iterating over all cell faces. Thus the size of the internalField in the NeoFOAM differs from that of OpenFOAM.
```
